howto/stream/exchange-oob-data.md

```diff
@@ -2,127 +2,151 @@
+
+1. Install the socat package:
+
+   ```
+   sudo apt install socat
+   ```
 
 1. Connect the Unix domain socket:
 
    ```
    socat - UNIX-CONNECT:/run/user/1000/anbox/sockets/webrtc_data_foo
    ```
 
-1. After the Unix domain socket is connected, type a message and hit the `Enter` key:
+1. This command establishes a connection to the `webrtc_data_foo` Unix domain socket, allowing you to send and receive data directly through the established data channel between Anbox runtime and the web client. After the Unix domain socket is connected, type a message and hit the `Enter` key:
 
        hello world
 
-   The data is now sent from the Anbox WebRTC platform over the data channel to the WebRTC client.
-1. Observe that the message is displayed in the console of a web browser, responding to the message event:
+   The data is now sent from the Anbox runtime over the data channel to the web client.
+1. Observe that the message is displayed in the [console](https://developer.chrome.com/docs/devtools/console) of a web client, responding to the message event:
 
        data received: hello world
 
-1. To test the other direction of the communication, send a message from a WebRTC client to the Anbox WebRTC platform through the data channel:
-
-   ```
-   session.sendData('foo', 'anbox cloud')
-   ```
+1. To test the other direction of the communication, in the web client, type a message and send the message over the `foo` data channel to the Anbox runtime.
 
 1. Observe that the received data is printed out in the `socat` TCP session:
 
    ```
    socat - UNIX-CONNECT:/run/user/1000/anbox/sockets/webrtc_data_foo
-   hello world <--  the sent data
-   anbox cloud <--  the received data
+   hello world
+   anbox cloud
    ```
 
-### Anbox WebRTC data proxy
+This enables data exchange between a service running on the Anbox instance and the web client. However, it does not yet facilitate data exchange between an Android application running inside the Android container and the web client.
+
+
+### Data exchange between Android application and web client
 
-To build up the communication bridge between an Android application and the Anbox WebRTC platform, Anbox Cloud provides a system daemon named `anbox-webrtc-data-proxy`. This daemon is responsible for:
+In the [Anbox Streaming SDK](https://github.com/canonical/anbox-streaming-sdk), there is an [out_of_band_v2](https://github.com/canonical/anbox-streaming-sdk/tree/master/examples/android/out_of_band_v2) project. You can either:
+- compile and modify the example application to meet your needs.
+- use the pre-built out-of-band v2 APK from the [release tarball](https://github.com/canonical/anbox-streaming-sdk/releases) to get started and immediately try out this feature by [running end-to-end tests](https://documentation.ubuntu.com/anbox-cloud/en/latest/howto/stream/exchange-oob-data/#run-end-to-end-test).
```

Comment:
Here we say they can use the example but below we talk again about another application ("Your Android application is required to implement a subclass of the ..."). Can't we just use only the example application we have built specifically for this purpose throughout this how-to and refer to parts of it for the explanation? I doubt that users will be able to implement their own app easily and will rather start right from the example for the first experiments. Wdyt?